versioned_hdf5/replay.py

```diff
@@ -117,7 +127,43 @@ def tmp_group(f):
         tmp = f['_version_data/__tmp__']
     return tmp
 
-def _recreate_raw_data(f, name, versions_to_delete, tmp=False):
+# See InMemoryDataset.fillvalue. In h5py3 variable length strings use None
+# for the h5py fillvalue, but require a string fillvalue for NumPy.
+def _get_np_fillvalue(data: Dataset) -> Union[bytes, float]:
```

Comment:
This is not strictly correct (the fillvalue can be any NumPy scalar type supported by h5py), but it doesn't really matter because type annotations aren't actually used anywhere.

Reply:
Hmm, maybe I'll just change this to `Any` for now.